packages/shared/stub-components.js

```diff
@@ -40,147 +25,72 @@ function getCoreProperties (component: Component): Object {
 }
 
-function stubComponents (components: Object, stubbedComponents: Object) {
-  Object.keys(components).forEach(component => {
-    // Remove cached constructor
-    delete components[component]._Ctor
-    if (!components[component].name) {
-      components[component].name = component
-    }
-    stubbedComponents[component] = createBlankStub(components[component])
-
-    // ignoreElements does not exist in Vue 2.0.x
-    if (Vue.config.ignoredElements) {
-      Vue.config.ignoredElements.push(component)
-    }
-  })
-}
-
-export function createComponentStubsForAll (component: Component): Object {
-  const stubbedComponents = {}
-
-  if (component.components) {
-    stubComponents(component.components, stubbedComponents)
+function normalizeStubOptions (components: ?Object, stubOptions: ?Object | Array<string>): Object {
+  if (!stubOptions) {
+    stubOptions = Object.keys(components || {})
   }
-
-  let extended = component.extends
-
-  // Loop through extended component chains to stub all child components
-  while (extended) {
-    if (extended.components) {
-      stubComponents(extended.components, stubbedComponents)
-    }
-    extended = extended.extends
+  if (Array.isArray(stubOptions)) {
+    stubOptions = stubOptions.reduce((object, name) => {
+      object[name] = true
+      return object
+    }, {})
   }
+  return stubOptions
+}
 
-  if (component.extendOptions && component.extendOptions.components) {
-    stubComponents(component.extendOptions.components, stubbedComponents)
-  }
+export function createComponentStubs (components: Object = {}, stubOptions: Object): Object {
+  validateStubOptions(stubOptions, components)
+  return createStubs(components, stubOptions)
+}
 
-  return stubbedComponents
+export function createComponentStubsForAll (component: Component, stubs: Object = {}): Object {
+  if (!component) return stubs
+  Object.assign(stubs, createStubs(component.components))
+  return createComponentStubsForAll(component.extends || component.extendOptions, stubs)
 }
 
-export function createComponentStubsForGlobals (instance: Component): Object {
-  const components = {}
-  Object.keys(instance.options.components).forEach((c) => {
-    if (isRequiredComponent(c)) {
-      return
-    }
+export function createStubs (components: Object, stubOptions: ?Object | Array<string>): Object {
+  const options: Object = normalizeStubOptions(components, stubOptions)
+
+  return Object.keys(options)
+    .filter(name => !['KeepAlive', 'Transition', 'TransitionGroup'].includes(name))
+    .filter(name => options[name] !== false)
+    .reduce((stubs, name) => {
+      stubs[name] = createStub(components[name], options[name])
 
-    components[c] = createBlankStub(instance.options.components[c])
-    delete instance.options.components[c]._Ctor // eslint-disable-line no-param-reassign
-    delete components[c]._Ctor // eslint-disable-line no-param-reassign
-  })
-  return components
+      if (Vue.config.ignoredElements) {
```

Comment:
Do we need it ? I copied it from original source.

Reply:
I don't think we do. you can remove it :)